source/globalCommands.py

```diff
@@ -1476,24 +1476,19 @@ def script_toggleProgressBarOutput(self,gesture):
 	script_toggleProgressBarOutput.category=SCRCAT_SPEECH
 
 	def script_togglePlayObjectCoordinates(self,gesture):
-		playObjectCoordinates = config.conf["presentation"]["playObjectCoordinates"]
-		if playObjectCoordinates == "both":
-			playObjectCoordinates = "off"
-			# Translators: A mode where object location coordinates will not be announced.
-			ui.message(_("Do not play Object location tone"))
-		elif playObjectCoordinates == "off":
-			playObjectCoordinates = "objNav"
-			# Translators: A mode where object coordinate beep will be heard when using object navigation commands.
-			ui.message(_("Play object location tone when using object navigation"))
-		elif playObjectCoordinates == "objNav":
-			playObjectCoordinates = "focus"
-			# Translators: A mode where object coordinate beep will be heard when moving system focus.
-			ui.message(_("Play object location tone when focus moves"))
-		else:
-			playObjectCoordinates = "both"
-			# Translators: A mode where object location beep will be heard always when moving focus and doing object navigation commands.
-			ui.message(_("Always play object location tone"))
-		config.conf["presentation"]["playObjectCoordinates"] = playObjectCoordinates
+		import NVDAObjects
+		values = [x[0] for x in NVDAObjects.objCoordinateChoices]
+		labels = [x[1] for x in NVDAObjects.objCoordinateChoices]
+		try:
+			index = values.index(config.conf["presentation"]["playObjectCoordinates"])
+		except:
+			index=0
+		newIndex = (index+1) % len(values)
+		config.conf["presentation"]["playObjectCoordinates"] = values[newIndex]
+		# Translators: Reports the new state of object coordinate playback.
+		# %s will be replaced with play object coordinates setting.
+		# For example, the full message might be "Play object coordinates: when focus moves"
+		ui.message(_("Play object coordinates: %s")%labels[newIndex].lower())

```

Comment:
I think this could be hard to translate in some languages. Why not have two sets of choices, one for here, and another for the settings dialog?

Reply:
Hi,

Ah that... This is the same approach taken by braille focus context options, and is also seen for symbol level command (NVDA+P). As for translations, it is really up to translators to decide how these strings should be translated after consulting with the community.

Thanks.